app/views/users/_form.html.erb

```diff
@@ -0,0 +1,45 @@
+<%= form_with model: user do |form| %>
+  <div>
+    <%= form.label :first_name %><br>
+    <%= form.text_field :first_name %>
+    <% user.errors.full_messages_for(:first_name).each do |message| %>
+      <div><%= message %></div>
+    <% end %>
+  </div>
+
+  <div>
+    <%= form.label :last_name %><br>
+    <%= form.text_field :last_name %>
+    <% user.errors.full_messages_for(:last_name).each do |message| %>
+      <div><%= message %></div>
+    <% end %>
+  </div>
+
+  <div>
+    <%= form.label :email %><br>
+    <%= form.text_field :email %>
+    <% user.errors.full_messages_for(:email).each do |message| %>
+      <div><%= message %></div>
+    <% end %>
+  </div>
+
+  <div>
+    <%= form.label :role %><br>
+    <%= form.text_field :role %>
+    <% user.errors.full_messages_for(:role).each do |message| %>
+      <div><%= message %></div>
+    <% end %>
+  </div>
+
+  <div>
+    <%= form.label :password %><br>
+    <%= form.text_field :password %>
+    <% user.errors.full_messages_for(:password).each do |message| %>
+      <div><%= message %></div>
+    <% end %>
+  </div>
+
+  <div>
+    <%= form.submit %>
+  </div>
+<% end %>
```

Comment:
BTW, we usually use `SimpleForm` gem (https://github.com/heartcombo/simple_form), so this file can be written in 4-5 lines :
```
<%= simple_form_for @user do |form| %>
  <%= form.input :username %>
  <%= form.input :password %>
  <%= form.input :role, as: :select %>
  <%= form.input :gender, as: :radio_buttons %>
  <%= form.button :submit %>
<% end %>
```
and this gem is well compatible with Bootstrap, so that building nice forms is incredibly simple...before you'll need a nice JS - than you life turned into the hell 😅 .

Reply:
Nice, thank you